### PR TITLE
Support overflowing health and mana

### PR DESCRIPTION
**Pull request by Incompleteusern**
image
image

### CODE REVIEW

File: common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java [hunk 1]
```diff
@@ -226,6 +226,45 @@ public BossBarModel.BarProgress progress() {
         }
 
         protected void renderBar(PoseStack poseStack, float renderY, float renderHeight, float progress) {
+            if (progress > 1) { // overflowing health
+                CustomColor yellow = CommonColors.YELLOW;
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java]: yellow. (unused variable)

File: common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java [hunk 2]
```diff
@@ -313,6 +352,43 @@ public boolean isActive() {
 
         @Override
         protected void renderBar(PoseStack poseStack, float renderY, float renderHeight, float progress) {
+            if (progress > 1) { // overflowing mana
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:353]: Bit of duplicate code tbh, do you think we can do anything with it? If not, this is fine I guess.

Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:353]: Maybe it is better not to abstract this into hell

File: common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java [hunk 3]
```diff
@@ -313,6 +352,43 @@ public boolean isActive() {
...
+                float x1 = this.getRenderX();
+                float x2 = this.getRenderX() + this.getWidth();
+                int textureY1 = manaTexture.getTextureY1();
+                int textureY2 = manaTexture.getTextureY2();
+
+                int half = (textureY1 + textureY2) / 2 + (textureY2 - textureY1) % 2;
+                RenderUtils.drawProgressBarBackground(
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:360]: Why do we need to do this?

File: common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java (hunk 2 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:353]: I think not abstracting is better

File: common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java (hunk 3 above)
Comment (Incompleteusern) [reply common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:360]: This is just the background of the bar

File: common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java (hunk 2 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:353]: Okay, I agree

File: common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java (hunk 3 above)
Comment (kristofbolyai) [reply common/src/main/java/com/wynntils/features/user/overlays/CustomBarsOverlayFeature.java:360]: Yeah right we return after this. Got it now

**Review by kristofbolyai — APPROVED** [review]

**Pull request merged**